### PR TITLE
Jtc sum periods

### PR DESCRIPTION
**Pull request by fmauch**
Sample trajectory based on the sum of periods instead of the absolute time.

This is basically the second part of #1191. In the long run I want to be able to scale the period with a scaling factor in order to  artificially slow down / speed up progression in the trajectory. This PR is there to separate the necessary time counting change from the scaling functionality to make things more contained and easier to review.

### CODE REVIEW

**Review by christophfroehlich — APPROVED** [review]
LGTM

File: joint_trajectory_controller/test/test_trajectory_actions.cpp [hunk 1]
```diff
@@ -83,12 +83,16 @@ class TestTrajectoryActions : public TrajectoryControllerTest
       {
         // controller hardware cycle update loop
         auto clock = rclcpp::Clock(RCL_STEADY_TIME);
-        auto start_time = clock.now();
+        auto now_time = clock.now();
+        auto start_time = now_time;
+        auto last_time = start_time;
         rclcpp::Duration wait = rclcpp::Duration::from_seconds(2.0);
         auto end_time = start_time + wait;
```
Comment (saikishor) [inline joint_trajectory_controller/test/test_trajectory_actions.cpp:90]:
```suggestion
        auto last_time = now_time;
        rclcpp::Duration wait = rclcpp::Duration::from_seconds(2.0);
        auto end_time = last_time + wait;
```

**Review by saikishor — APPROVED** [review]
Looks good to me. 

Doing this way, we are kinda doing it open loop with the time. Am I right?

**Comment by fmauch** [comment]
> Looks good to me.
> 
> Doing this way, we are kinda doing it open loop with the time. Am I right?

Yes, that is true. That requires correct period times without missing or skipping a period. Which is why we recently found issues with that and fixed them. From what I can draw from the design concepts this is an assumption we can make here, though.

**Comment by saikishor** [comment]
> Yes, that is true. That requires correct period times without missing or skipping a period. Which is why we recently found issues with that and fixed them. From what I can draw from the design concepts this is an assumption we can make here, though.

Got it. Thanks for the explanation.

**Comment by fmauch** [comment]
Semi-binary builds should now also succeed, now that https://github.com/ros-controls/ros2_control/pull/1898 is merged. I cannot restart the checks, though.

**Pull request merged**